cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/InMemoryDatasetOpExecutor.java

```diff
@@ -61,7 +61,9 @@ public DatasetSpecification create(Id.DatasetInstance datasetInstanceId, Dataset
 
     // TODO: Note - for now, just sending the name. However, type likely needs to be namesapce-aware too.
     DatasetSpecification spec = type.configure(datasetInstanceId.getId(), props);
-    spec = spec.setDescription(props.getDescription());
+    if (props.getDescription() != null) {
+      spec = spec.setDescription(props.getDescription());
+    }
```

Comment:
There is  DatasetDefintion.configure() and DatasetType.configure() (which delegates to the former)